### ISSUE TITLE
Add matchOne to EntitySpecificationRepositoryTrait

### ISSUE THREAD

**Issue by adamquaile** [Happyr/Doctrine-Specification#110]
I was thinking a `matchOne($spec)` method in `EntitySpecificationRepositoryTrait` would be useful, throwing an exception if 0 or >1 results are found. 

I would PR this if you think it's useful?

**Comment by drewsonian** [comment]
I would find that useful, as I'm manually checking for that scenario in several cases at this point.  An exception would be easier to handle IMHO.

**Comment by Nyholm** [comment]
Thank you for this suggestion. I like the idea. 
If I'm not misremembering there are a way of achieving this. Does the behavior for matchOne differ from using the following?

```php
// Will throw a NoResultException if nothing found
$oneResult = $repo->match(Spec::andX(
  new MySpec(),
  Spec::limit(1)
), Spec::asSingle());
```

**Comment by adamquaile** [comment]
I didn't know about that. That would do the trick, except that `AsSingle` uses `Query::HYDRATE_SINGLE_SCALAR` so it throws a `Doctrine\ORM\NonUniqueResultException` for objects.

Changing the hydration mode would be a BC breaking change. Perhaps `Spec::asSingle(AsSingle::OBJECT)`, but then the API is starting to look a bit untidy. What are your thoughts?

**Comment by Nyholm** [comment]
Lets introduce `Spec::asSingleScalar` and rewrite `Spec::asSingle`. Would that work? 
Please send a PR if you want to.

**Comment by adamquaile** [comment]
Yes, that would work. Would you also want `matchOne` as syntactic sugar for `Spec::asSingle`, or just the one method?

**Comment by Nyholm** [comment]
Im not too sure if about introducing `matchOne` on the `EntitySpecificationRepositoryTrait`. I want to try to solve problems using Specifications.

**Comment by adamquaile** [comment]
Okay, no problem.

**Comment by Nyholm** [comment]
Fixed by #113

**Comment by Nyholm** [comment]
I've tagged version 0.6.1 which includes this feature.
